### ISSUE TITLE
Can't edit text due to an attempt to load non-existent js file, but only when running in production (port 80)

### ISSUE THREAD

**Issue by contextref** [divio/djangocms-text-ckeditor#438]
Very strange: when I am running on a server with port 8000, all is well.  When in production on port 80 (or localhost port 80), I am getting this 404 error when trying to edit a text plugin:

[26/Aug/2017 20:35:20] "GET /static/djangocms_text_ckeditor/ckeditor/plugins/copyformatting/lang/en.js?t=H0CG HTTP/1.1" 404 147

As a result, text editor does not load properly and text can't be edited.

this file does not exist at the path given above, and when running on :8000, there is no request for that URL at all.  Any ideas?....
Update: I stumbled onto this: https://stackoverflow.com/questions/15786690/django-escaping-javascript-when-debug-false and tried the workaround they suggested, and it helped, which was doing this in settings:

`EXCLUDE_FROM_MINIFYING = ('^admin/', '^en/admin/')
`

**Comment by vxsx** [comment]
Closing, since not a bug with djangocms-text-ckeditor itself, but thanks for posting, hope it'll help someone!

**Comment by debuggerpk** [comment]
i am getting the same error

**Comment by debuggerpk** [comment]
figured out the error. Its the way static files were setup. The best possible solution to get rid of the error is to do

```bash
./manage.py collectstatic -c
```

the -c flag would clean the static files first. The above is a dangerous command. So make sure the project STATICFILES_DIR is setup correctly, or else it will clean everything and then do the setup.

**Comment by krisb78** [comment]
Having the same error, in my case the missing file is:

/static/djangocms_text_ckeditor/ckeditor/plugins/copyformatting/lang/pl.js?t=H8DA

I looked at the s3 bucket where my static files are collected, this file is missing.

It is not being requested when I'm running in the Debug mode (the editor actually loads then).

Looking at the static folder in this project, this file is not there, either.

Is there some setting I'm missing or have misconfigured?

**Comment by darbula** [comment]
Same here. This is not a good solution but it helps:

I manually copied `https://github.com/ckeditor/ckeditor-dev/tree/4.7.3/plugins/copyformatting/lang` folder to corresponding static folder of this project.

@vxsx is there any reason why those files are missing from this repo?

**Comment by vxsx** [comment]
@darbula because we don't use development version of ckeditor, we use a compiled one, so the translations for this plugin are here: https://github.com/divio/djangocms-text-ckeditor/blob/master/djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor/lang/pl.js

**Comment by darbula** [comment]
@vxsx thank you, it seems that now that file is not requested anymore.

**Comment by krisb78** [comment]
@vxsx why would the browser request these files then? This error happens with `DEBUG = False`, is there a specific setting in djangocms-text-ckeditor to control the version of ckeditor?

**Comment by vxsx** [comment]
@krisb78 I have no idea, haven't seen it before. Normally version of ckeditor is controlled by https://github.com/divio/djangocms-text-ckeditor/blob/master/djangocms_text_ckeditor/widgets.py#L19 

errors may happen if you upgrade from old version where widget html contained a reference to a file and the html file is still in the project

i would have to debug the js calls leading to incorrect requests to figure out what is the problem

**Comment by krisb78** [comment]
@vxsx I finally found the root cause of the issue. I have my static assets collected to s3, and file:
`djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor/lang/pl.js` was not getting updated in s3 I upgraded the ckeditor plugin, because the `collectstatic` command didn't believe it has changed.

`collectstatic` compares mtime of files to determine whether to update them or not, and there is something wrong with the logic there, at least when s3 is used for staticfiles storage.

This is why `./manage.py collectstatic -c` solves this.
